src/if_statements.py

```diff
@@ -353,55 +288,208 @@ def immediate_postdominator(context: Context, start: Node, end: Node) -> Node:
     return postdominators[0]
 
 
-def count_non_postdominated_parents(
-    context: Context, child: Node, curr_end: Node
-) -> int:
+def is_and_statement(next_node: Node, bottom: Node) -> bool:
+    return (
+        not isinstance(next_node, ConditionalNode)
+        or next_node not in bottom.parents
+        # A strange edge-case of our pattern-matching technology:
+        # self-loops match the pattern. Avoiding that...
+        or next_node.is_loop()
```

Comment:
can you explain this case further? if this method is inlined into get_full_if_condition the comment may find a more natural place to live